### PR TITLE
(feat): Combine common Transmitter into FunctionsContractTransmitter

### PR DESCRIPTION
**Pull request by justinkaseman**

### CODE REVIEW

File: core/services/ocrcommon/transmitter.go [hunk 1]
```diff
@@ -71,11 +71,18 @@ func (t *transmitter) CreateEthTransaction(ctx context.Context, toAddress common
 		return errors.Wrap(err, "skipped OCR transmission, error getting round-robin address")
 	}
 
+	var gasLimit uint64
+	if txMeta != nil && txMeta.GasLimit != nil {
```
Comment (prashantkumar1982) [inline core/services/ocrcommon/transmitter.go:75]: Who sets this field "txMeta.GasLimit"?
Will it be added in a later PR?

Comment (justinkaseman) [reply core/services/ocrcommon/transmitter.go:75]: Yes exactly. The place where Functions would use this is here:
https://github.com/smartcontractkit/chainlink/blob/30b73a804dfba394180abe354569dade80a71be5/core/services/relay/evm/functions/contract_transmitter.go#L164

File: core/services/relay/evm/functions/contract_transmitter.go [hunk 2]
```diff
@@ -87,18 +110,58 @@ func NewFunctionsContractTransmitter(
...
+func (oc *contractTransmitter) forwarderAddress() common.Address {
+	for _, a := range oc.fromAddresses {
+		if a == oc.effectiveTransmitterAddress {
+			return common.Address{}
+		}
+	}
+	return oc.effectiveTransmitterAddress
+}
+
+func (oc *contractTransmitter) fromAddress() common.Address {
```
Comment (bolekk) [inline core/services/relay/evm/functions/contract_transmitter.go:161]: we don't need a helper if it's used only once

File: core/services/relay/evm/functions/contract_transmitter.go [hunk 3]
```diff
@@ -25,33 +26,43 @@ import (
...
-	reportToEvmTxMeta   ReportToEthMetadata
-	contractVersion     uint32
-	reportCodec         encoding.ReportCodec
+	contractAddress             atomic.Pointer[common.Address]
+	contractABI                 abi.ABI
+	transmittedEventSig         common.Hash
+	contractReader              contractReader
+	lp                          logpoller.LogPoller
+	lggr                        logger.Logger
+	reportToEvmTxMeta           ReportToEthMetadata
```
Comment (bolekk) [inline core/services/relay/evm/functions/contract_transmitter.go:55]: We can get rid of this, it's always set to nil (= no-op). Also remove reportToEvmTxMetaNoop function.

File: core/services/relay/evm/functions/contract_transmitter.go [hunk 4]
```diff
@@ -87,18 +110,54 @@ func NewFunctionsContractTransmitter(
...
+		gasLimit:                    gasLimit,
+		effectiveTransmitterAddress: effectiveTransmitterAddress,
+		strategy:                    strategy,
+		checker:                     checker,
+		chainID:                     chainID,
+		keystore:                    keystore,
 	}, nil
 }
 
+func (oc *contractTransmitter) createEthTransaction(ctx context.Context, toAddress common.Address, payload []byte, txMeta *txmgr.TxMeta) error {
```
Comment (bolekk) [inline core/services/relay/evm/functions/contract_transmitter.go:132]: I think txMeta is always nil so we could remove it from params.

File: core/services/relay/evm/functions/contract_transmitter.go [hunk 5]
```diff
@@ -87,18 +110,54 @@ func NewFunctionsContractTransmitter(
...
-		contractReader:      caller,
-		lggr:                lggr.Named("OCRContractTransmitter"),
-		reportToEvmTxMeta:   reportToEvmTxMeta,
-		contractVersion:     contractVersion,
-		reportCodec:         codec,
+		contractABI:                 contractABI,
+		transmittedEventSig:         transmitted.ID,
+		lp:                          lp,
+		contractReader:              caller,
+		lggr:                        lggr.Named("OCRContractTransmitter"),
```
Comment (agparadiso) [inline core/services/relay/evm/functions/contract_transmitter.go:117]: would it help when debugging if the lggr was named: `OCRFunctionsContractTransmitter`?

**Review by agparadiso — APPROVED** [review]

**Review by bolekk — APPROVED** [review]

**Pull request merged**